### ISSUE TITLE
waterline doesn't cast criteria object against model schema

### ISSUE THREAD

**Issue by clarkorz** [balderdashy/waterline#505]
Because `sails` pass criteria object to `waterline` directly, all value part of key-pair clauses in criteria are `string` type. e.g.:

    http://localhost:1337/users?age[>]=10&age[<]=30

will parse as waterline criteria:

    { age: { '>': '10', '<': '30' } }

will parse as mongo criteria:

    { age: { $gt: '10', $lt: '30' } }

But `age` is `integer` type in both model definition and database, hence this query won't get any result.

Currently, I have to custom almost every controller actions that involve with querying on non-string value, and cast every possible value manually.

**Comment by dmarcelino** [comment]
@clarkorz, is this still an issue? Can you provide a test? Thanks

**Comment by devinivy** [comment]
@particlebanana at what point would criteria be cast?  Off the top of my head, I can't think of anywhere this happens in waterline core.

**Comment by dmarcelino** [comment]
@devinivy, good point, I know there is some casting going on at [typecast.js](https://github.com/balderdashy/waterline/blob/master/lib/waterline/core/typecast.js) which applies to values. Not sure if this is used (or can be used) for criteria. It would make some sense for us to cast criteria and values consistently.

**Comment by sailsbot** [comment]
Thanks for posting, @clarkorz.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
